tests/utils/verification/volumeproperties.go

```diff
@@ -152,6 +155,29 @@ func VerifyDetachedStatus(name, hostName, esxName string) bool {
 	return false
 }
 
+// VerifyDetachedStatusOnNonDefaultDS - For volume on non-default DS,
+// check if the status gets detached within the timeout
+func VerifyDetachedStatusOnNonDefaultDS(name, datastore, hostName, esxName string) bool {
+	log.Printf("Confirming detached status for volume [%s] on datastore[%s]\n", name, datastore)
+
+	//TODO: Need to implement generic polling logic for better reuse
+	const maxAttempt = 60
+	for attempt := 0; attempt < maxAttempt; attempt++ {
+		misc.SleepForSec(2)
+		status := getVolumeStatusHost(name+"@"+datastore, hostName)
```

Comment:
Agree with @pshahzeb. I think we can do the same thing as what we are doing in VerifyAttachedStatus - please refer to issue #1506 and the fix: https://github.com/vmware/docker-volume-vsphere/pull/1492/commits/54ea0f1120db6ab683b83f4d8f6e033cfd9273a6